src/VAC/View.cpp

```diff
@@ -366,17 +373,40 @@ int View::decidePMRAction()
         }
         else if(global()->toolMode() == Global::DRAW_LINE)
         {
-          return LINE_ACTION;
+            return LINE_ACTION;
         }
         else if(global()->toolMode() == Global::DRAW_RECTANGLE)
         {
-          return RECTANGLE_ACTION;
+            return RECTANGLE_ACTION;
         }
-        else if(global()->toolMode() == Global::DRAW_CIRCLE)
+            else if(global()->toolMode() == Global::DRAW_CIRCLE)
         {
-          return CIRCLE_ACTION;
+            return CIRCLE_ACTION;
+        }
+        else if(global()->toolMode() == Global::DRAW_TRIANGLE)
+        {
+            return TRIANGLE_ACTION;
+        }
+        else if(global()->toolMode() == Global::DRAW_RHOMBUS)
+        {
+            return RHOMBUS_ACTION;
+        }
+        else if(global()->toolMode() == Global::DRAW_PENTAGON)
+        {
+            return PENTAGON_ACTION;
+        }
+        else if(global()->toolMode() == Global::DRAW_HEXAGON)
+        {
+            return HEXAGON_ACTION;
+        }
+        else if(global()->toolMode() == Global::DRAW_HEPTAGON)
+        {
+            return HEPTAGON_ACTION;
+        }
+        else if(global()->toolMode() == Global::DRAW_OCTAGON)
+        {
+            return OCTAGON_ACTION;
```

Comment:
maybe use switch-case or a QHash to do this mapping?  :)

Reply:
I agree.
But maybe it would be better if I will do this refactoring in the future in next PRs, because it is only part of big if-ele-if and in View.cpp are a lot of places when it would be better to use switch-case instead if-else-if. And I would like to perform the some additional optimizations and improvements also. Also I plan try to implement real time preview during drawing shapes. I would like to focus on implementing maximum num of functional for the sprint preview presentation.